### PR TITLE
chore: gatsby2 migration

### LINKED ISSUE #139: Migrate to Gatsby 2
Gatsby 2 is currently in beta but I think it's worth to migrate:

- Better webpack performance
- Smaller bundles
- We can use dynamic imports to lazy-load sections, see #69 
- More future proof

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR migrates the project from [gatsbyjs v1](https://v1.gatsbyjs.org/) to [gatsbyjs v2](https://www.gatsbyjs.org/). 

### Important changes:

- Updated `gatsbyjs` version
- Updated `gatsbyjs` related packages
- Updated `gatsbyjs` plugins' peer dependencies:
   - [gatsby-link](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-link)
   - [gatsby-plugin-react-helmet](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-react-helmet)
- layouts (See [ref](https://www.gatsbyjs.org/docs/migrating-from-v1-to-v2/#remove-or-refactor-layout-components))
   - changed `children` `render prop` to a normal `prop`;
   - updated `layout` component to receive `className` as a `prop`
   - there are no multiple layouts anymore (one for each page) but instead, just one layout component is applied to each page and it renders `props.children`
   - we've decided to use [gatsby-plugin-layout](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-layout/) for two reasons: 
        1. keep the same structure of the previous Gatsbyjs v1 where the layout is located at `src/layouts`. 
        2. the current hierarchy is `Page -> Layout`, which differs from the old one that was `Layout -> Page`. This means that with the current hierarchy, everytime that a navigation between pages occurs, react will unmount the old page and mount the new one. Hence, everything inside `Page` also unmounts. With this plugin, we can still have the old hierarchy: `Layout -> Page`.
- Updated `gatsby-node.js` according its [new API](https://www.gatsbyjs.org/docs/node-apis/) and changed webpack config according to webpack 4.

This PR is an improvement of PR #191 as we change our layout approach. 
Closes #139.

### CODE REVIEW

File: src/layouts/index.js [hunk 1]
```diff
@@ -39,17 +47,28 @@ class Layout extends Component {
...
+      this.previousIntl = intl
+      loadLocaleData(intl)
+    }
+  }
 }
 
 Layout.propTypes = {
-  children: PropTypes.func
+  children: PropTypes.object,
+  className: PropTypes.string,
```
Comment (satazor) [inline src/layouts/index.js]: I think we can revert the classNamd property, we don’t use it.

File: gatsby-config.js [hunk 2]
```diff
@@ -27,12 +29,6 @@ module.exports = {
           }
         ]
       }
-    },
```
Comment (satazor) [inline gatsby-config.js]: Why did we remove this?

File: gatsby-node.js [hunk 3]
```diff
@@ -8,229 +8,178 @@
...
-module.exports.modifyWebpackConfig = ({ config, program, stage }) => {
-  // Allow requires from the src/ folder
-  config.merge({
-    resolve: {
-      root: path.join(__dirname, 'src')
-    },
-    target: 'web'
-  })
-
-  // Fix Gatsby setting `resolve.modulesDirectories` to `path.resolve(__dirname, "node_modules")`
```
Comment (satazor) [inline gatsby-node.js]: I think this comment is better than the one we have now :D

File: gatsby-node.js [hunk 4]
```diff
@@ -8,229 +8,178 @@
...
+  // See https://github.com/gatsbyjs/gatsby/blob/a0cbbcb5519e53d2358a06be2c1a6ca0688280b7/packages/gatsby/src/utils/webpack.config.js#L363
+  // Furthermore, we add the `src` folder so that we can import internal modules without having to use relative paths
+  config.resolve.modules = ['node_modules', path.join(__dirname, 'src')]
 
-      return current
-    })
+  config.resolve.alias = {
+    ...config.resolve.alias,
+    // Add an alias to 'intl' so that we can import the intl config from within the src/
+    intl: path.join(__dirname, 'intl')
```
Comment (satazor) [inline gatsby-node.js]: Can we get rid of this and use `shared/intl-config` like before? I think it was more obvious. It was probably me who said to change this but nvm me :D

File: gatsby-node.js (hunk 3 above)
Comment (satazor) [reply gatsby-node.js]: But keep the current phrase about the `src/` folder

File: gatsby-node.js [hunk 5]
```diff
@@ -8,229 +8,172 @@
...
-    },
-    target: 'web'
-  })
+const SvgStorePlugin = require('external-svg-sprite-loader')
+const { defaultLocale, availableLocales } = require('./intl/config')
+
+module.exports.onCreateWebpackConfig = ({ actions, getConfig, stage }) => {
+  let config = getConfig()
 
   // Fix Gatsby setting `resolve.modulesDirectories` to `path.resolve(__dirname, "node_modules")`
```
Comment (satazor) [inline gatsby-node.js]: change `resolve.modulesDirectories` to `resolve.modules`
change ``path.resolve(__dirname, "node_modules")` to just `node_modules`

Comment (PedroMiguelSS) [reply gatsby-node.js]: Sorry, missed that detail.

**Pull request merged**